### DISCUSSION TITLE
msgpack pickler

### DISCUSSION THREAD

**Discussion by earonesty** [msgpack/msgpack-python#588] [General]
hi, i wrote this:

https://github.com/earonesty/msgpickle

it's a pickler that can be made safe, explicit and works with your classes

explicity register serializers for classes, or you use `to_pack` and `from_pack` handlers on your class

uses msgpack as the base serializaton layer.   very configurable.  can pack/unpack lambdas (this isnt safe, but its fun)

**Comment by ThomasWaldmann** [comment]
@earonesty That looks interesting!

Guess I have to check within `borgbackup` whether we would like to use your code instead of some homegrown stuff.
